src/test/basics/Buffer_test.cpp

```diff
@@ -0,0 +1,240 @@
+        }
+
+        // Check move constructor & move assignments:
+        {
+            testcase ("Move Construction / Assignment");
+
+            { // Move-construct from empty buf
+                Buffer x;
+                Buffer y { std::move(x) };
+                BEAST_EXPECT (x.empty());
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (y.empty());
+                BEAST_EXPECT (sane(y));
+                BEAST_EXPECT (x == y);
+            }
+
+            { // Move-construct from non-empty buf
+                Buffer x { b1 };
+                Buffer y { std::move(x) };
+                BEAST_EXPECT (x.empty());
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (y == b1);
+            }
+
+            { // Move assign empty buf to empty buf
+                Buffer x;
+                Buffer y;
+
+                x = std::move(y);
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (x.empty());
+                BEAST_EXPECT (sane(y));
+                BEAST_EXPECT (y.empty());
+            }
+
+            { // Move assign non-empty buf to empty buf
+                Buffer x;
+                Buffer y { b1 };
+
+                x = std::move(y);
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (x == b1);
+                BEAST_EXPECT (sane(y));
+                BEAST_EXPECT (y.empty());
+            }
+
+            { // Move assign empty buf to non-empty buf
+                Buffer x { b1 };
+                Buffer y;
+
+                x = std::move(y);
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (x.empty());
+                BEAST_EXPECT (sane(y));
+                BEAST_EXPECT (y.empty());
+            }
+
+            { // Move assign non-empty buf to non-empty buf
+                Buffer x { b1 };
+                Buffer y { b2 };
+                Buffer z { b3 };
+
+                x = std::move(y);
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (!x.empty());
+                BEAST_EXPECT (sane(y));
+                BEAST_EXPECT (y.empty());
+
+                x = std::move(z);
+                BEAST_EXPECT (sane(x));
+                BEAST_EXPECT (!x.empty());
+                BEAST_EXPECT (sane(z));
+                BEAST_EXPECT (z.empty());
+            }
+        }
+
+        {
+            testcase ("Slice Conversion / Construction / Assignment");
+
+            Buffer w { static_cast<Slice>(b0) };
+            BEAST_EXPECT(sane(w));
+            BEAST_EXPECT(w == b0);
+
+            Buffer x { static_cast<Slice>(b1) };
+            BEAST_EXPECT(sane(x));
+            BEAST_EXPECT(x == b1);
+
+            Buffer y { static_cast<Slice>(b2) };
+            BEAST_EXPECT(sane(y));
+            BEAST_EXPECT(y == b2);
+
+            Buffer z { static_cast<Slice>(b3) };
+            BEAST_EXPECT(sane(z));
+            BEAST_EXPECT(z == b3);
+
+            // Assign empty slice to empty slice
```

Comment:
"Assign empty slice to empty buffer"?  Similar comment fixes through the rest of this section I believe.